plugins/solaris/io_disk

```diff
@@ -0,0 +1,301 @@
+        conf_vlabel='Queue length wait (-) / actv (+)'
+        ;;
+    latency)
+        conf_title='Latency'
+        conf_in=rlentime
+        conf_out=wlentime
+        conf_cdef=
+        conf_gargs='--base 1000'
+        conf_vlabel='Seconds wsvc_t (-) / asvc_t (+)'
+        ;;
+    *)
+        echo "Unknown function: $func"
+        exit 1
+        ;;
+    esac
+}
+
+is_excluded() {
+    local dev i
+    dev=$1
+
+    for i in ${exclude}
+    do
+        if [ "$dev" = "$i" ]; then
+            return 0
+        fi
+    done
+
+    return 1
+}
+
+do_config() {
+    local func dev ref devname stat
+
+    func=$1
+    preconfig "$func"
+
+    echo "multigraph ${plugin_name}_${func}_${module}"
+
+    echo "graph_title $title_type $conf_title"
+    echo 'graph_category disk'
+    echo "graph_args $conf_gargs"
+    echo "graph_vlabel $conf_vlabel"
+    if [ -n "$graph_width" ]; then
+        echo "graph_width $graph_width"
+    fi
+
+    kstat -p -c "$class" -m "$module_regex" -s "/^${conf_in}\$/" | sed 's/:/ /g' | awk '{ print $3 }' | \
+    while read -r dev
+    do
+        if is_excluded "$dev" ; then
+            continue
+        fi
+
+        # replace instance name to logical device name if exists
+        ref="name_$dev"
+        devname=${!ref:-}
+
+        # reads and writes are necessary to calculate latency
+        if [ "$func" = 'latency' ]; then
+            for stat in reads writes
+            do
+                echo "${dev}_$stat.label dummy" 
+                echo "${dev}_$stat.graph no"
+                echo "${dev}_$stat.type DERIVE"
+                echo "${dev}_$stat.min 0"
+            done
+            # rlentime / ( reads + writes )
+            conf_cdef=,${dev}_reads,${dev}_writes,+,/
+        fi
+
+        echo "${dev}_${conf_out}.label dummy" 
+        echo "${dev}_${conf_out}.graph no"
+        echo "${dev}_${conf_out}.type DERIVE"
+        echo "${dev}_${conf_out}.min 0"
+
+        echo "${dev}_${conf_in}.label ${devname:-${dev}}"
+        echo "${dev}_${conf_in}.negative ${dev}_${conf_out}"
+        echo "${dev}_${conf_in}.type DERIVE"
+        echo "${dev}_${conf_in}.min 0"
+
+        if [ -n "$conf_cdef" ]; then
+            echo "${dev}_${conf_out}.cdef ${dev}_${conf_out}${conf_cdef}"
+            echo "${dev}_${conf_in}.cdef ${dev}_${conf_in}${conf_cdef}"
+        fi
+    done
+
+    echo
+}
+
+do_getvalue() {
+    local func stat_regex dev stat value
+
+    func=$1
+    preconfig "$func"
+
+    if [ "$func" = 'latency' ]; then
+        stat_regex="/^($conf_in|$conf_out|reads|writes)\$/"
+    else
+        stat_regex="/^($conf_in|$conf_out)\$/"
+    fi
+
+    echo "multigraph ${plugin_name}_${func}_${module}"
+
+    kstat -p -c "$class" -m "$module_regex" -s "$stat_regex" | sed 's/:/ /g' | awk '{ print $3,$4,$5 }' | \
+    while read -r dev stat value
+    do
+        if is_excluded "$dev" ; then
+            continue
+        fi
+
+        echo "${dev}_${stat}.value ${value%.*}"
```

Comment:
Yes, I do. I think rrdtool can accept only integers, can't it? Since, cumulative I/O ops and I/O bytes are quite large values (like millions), I think discard precision does not really matter.